src/pages/ExerciseDetails/index.tsx

```diff
@@ -35,23 +36,29 @@ export const ExerciseDetails = () => {
         // set to state so it can be rendered directly
         try {
             const exerciseReceived = await getExerciseBase(exerciseID);
+            const languages = await getLanguages();
+            const variantExercises = await getExerciseBases();
             //collect user browser's language
-            const currentUserLanguage = getLanguageByShortName(i18n.language, state.languages);
+            const currentUserLanguage = getLanguageByShortName(i18n.language, languages);
             // get exercise translation from received exercise and set it
             if (currentUserLanguage) {
                 const newTranslatedExercise = exerciseReceived?.getTranslation(currentUserLanguage);
                 setCurrentTranslation(newTranslatedExercise);
             }
             setCurrentUserLanguageState(currentUserLanguage);          
             setExerciseState(exerciseReceived);
+            dispatch(setLanguages(languages));
+            // slice the first 3 exercises to display in variants
+            // It's been set in the global state thogh
+            dispatch(setExerciseBases(variantExercises.slice(0, 3)));
```

Comment:
Note, we should show all possible variants, not just the first three (even if we still have to collect this information)